File: app/src/main/java/me/zipi/navitotesla/model/ShareRequest.java
package me.zipi.navitotesla.model;

import com.google.gson.annotations.SerializedName;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
public class ShareRequest {
    String type;
    String locale;
    @SerializedName("timestamp_ms")
    Long timestampMs;
    Value value;

    public ShareRequest(String address) {
        this.timestampMs = System.currentTimeMillis() / 1000L;
        this.type = "share_ext_content_raw";
        this.locale = "ko-KR";
        this.value = new Value(address);
    }

    @Data
    @NoArgsConstructor
    @AllArgsConstructor
    public static class Value {
        @SerializedName("android.intent.extra.TEXT")
        String text;
    }

}

File: app/src/main/kotlin/me/zipi/navitotesla/model/ShareRequest.kt
package me.zipi.navitotesla.model

import com.google.gson.annotations.SerializedName

data class ShareRequest(
    val address: String,
) {
    val type = "share_ext_content_raw"
    val locale = "ko-KR"

    @SerializedName("timestamp_ms")
    val timestampMs: Long = System.currentTimeMillis() / 1000L
    val value: Value = Value(address)

    data class Value(
        @SerializedName("android.intent.extra.TEXT")
        val text: String,
    )
}
/**
 * POST https://owner-api.teslamotors.com/api/1/vehicles/1493132499069707/command/share
 *
 * {
 *     "locale": "ko-KR",
 *     "timestamp": 1704673330414,
 *     "type": "share_ext_content_raw",
 *     "value": {
 *         "android.intent.ACTION": "android.intent.action.SEND",
 *         "android.intent.TYPE": "text/plain",
 *         "android.intent.extra.TEXT": "37.3284973,127.112211"
 *     }
 * }
 */
